### ISSUE TITLE
count() errors

### ISSUE THREAD

**Issue by marcalj** [creatoro/jelly#58]
I find a bug when updating User data with Auth module.

When I try to update the user like this:
`$this->user->update_user($_POST, array('username', 'password', 'email'));`

It throws an error when `_is_unique` is call. classes/jelly/cire/field: 279

```
Unknown column 'email@test.com' in 'where clause' ( SELECT COUNT(*) AS `total` FROM `users` AS `users` WHERE `email@test`.`com` = ':key' LIMIT 1 )
```

Kohana-3.1.2, Jelly-latest

**Comment by leth** [comment]
It looks like you've got the keys and values arrays the wrong way around.

If this isn't the problem, can you also post your "update_user" function?

**Comment by creatoro** [comment]
Are you sure you are using the latest version of Jelly (3.1/develop branch)? The `is_unique()` method has been updated since it was using the `:key` wildcard.

**Comment by marcalj** [comment]
Yes, latest code. Th update function is the default on classes/model/auth/user.php

```
    public function update_user($values, $expected)
    {
        if (empty($values['password']))
        {
            unset($values['password'], $values['password_confirm']);
        }

        // Validation for passwords
        $extra_validation = Model_User::get_password_validation($values);

        return $this->set(Arr::extract($values, $expected))->save();
    }
```

**Comment by leth** [comment]
Oh sorry, I missed the bit about the Auth module.

**Comment by marcalj** [comment]
I think the error is here, but i'm not sure:
https://github.com/creatoro/kohana-jelly-for-Kohana-3.1/commit/8b0fe01e7cea7651c1ea4568d4ba3f0332741f3e

**Comment by marcalj** [comment]
More light. The Jelly_Validation seems to correctly set email data:

```
        <small>protected</small> _bound => <small>array</small><span>(4)</span> <span>(
            ":model" => <small>object</small> <span>Model_User(10)</span> {
                *RECURSION*
            }
            ":validation" => <small>object</small> <span>Jelly_Validation(5)</span> {
                *RECURSION*
            }
            ":field" => <small>string</small><span>(5)</span> "email"
            ":value" => <small>string</small><span>(25)</span> "email@test.com"
        )</span>
```

But this mapping is failing:

```
        // Check as to whether we need to add
        // some callbacks for shortcut properties
        if ($this->unique === TRUE)
        {
            $this->rules[] = array(array(':field', '_is_unique'), array(':validation', ':model', ':field', ':value'));
        }
```

Because you get:

```
Kohana::$log->add(Log::INFO, Debug::vars($field, $value));
<small>string</small><span>(25)</span> "email@test.com"
<small>string</small><span>(4)</span> ":key"</pre>
```

**Comment by creatoro** [comment]
I've tested the update method and it worked for me without any problems.
But still, the `:key` wildcard doesn't seem right. Where is that coming from?

**Comment by marcalj** [comment]
I don't know, seems an error when parsing the rules... I'm using Kohana 3.1.2, can be this the problem?

**Comment by marcalj** [comment]
Hi, I've update Kohana to 3.1.3.1 and this problem disappears. So, please, tell in README and description that current code requires at lease v3.1.3 of Kohana framework.

Also I found a bug in the update_user function, here is the fixed code:

```
    public function update_user($values, $expected)
    {
        if (empty($values['password']))
        {
            unset($values['password'], $values['password_confirm']);
            unset($expected['password']);
        }

        $this->meta()->field('password')->rules = array(
//          array('not_empty'),
            array('min_length', array(':value', 8)),
        );

        // Validation for passwords
        $extra_validation = Model_User::get_password_validation($values);

        return $this->set(Arr::extract($values, $expected))->save($extra_validation);
    }
```

It's really hacky. I think that `$extra_validation` isn't overwrite the "password" field rules.

**Comment by creatoro** [comment]
@marcalj: I've created a separate issue for the Auth driver, and made a fix: 4995eb5ea0d5fb7b6677
Was that what you were talking about?

**Comment by marcalj** [comment]
Thanks, I'll comment on the issue.
